configure.ac

```diff
@@ -444,10 +444,20 @@ if test "x$with_dl" != "xno"; then
 fi
 
 ## check for atomics library before anything that might need it
-# AC_SEARCH_LIBS pollutes LIBS
 SQUID_STATE_SAVE(LIBATOMIC)
-AC_SEARCH_LIBS([__atomic_load_8],[atomic],[
-  test "$ac_res" = "none required" || ATOMICLIB=$ac_res],[])
+AC_MSG_CHECKING(library containing __atomic_is_lock_free)
+LIBS="-latomic"
+AC_LINK_IFELSE(
+  [AC_LANG_SOURCE([[
```

Comment:
I would expect us to supply standard headers and use a standard C++ interface instead of an "internal" GCC function with two leading underscores. Is there something that forces us to use GCC `__atomic` builtins here?